### 683380576-2 นายธนธรณ์ ผาลัง

## Lab 1.1: Environment Setup & First Script

In [ ]:
print("Hello Python Course")

Hello Python Course


### Lab 1.2: Working with Variables and Basic Data Types
- Objective: Students will define variables of different data types, perform simple operations, and display results.


2. Declare and print variables:


In [ ]:
quantity = 10
price = 195.5
title = "Basic Python 101"
available = True

print("Book title : ", title, "\t" ,type(title))
print("Book available : ", available, "\t\t", type(available))
print("Book quantity : ", quantity, "\t\t", type(quantity))
print("Book price : ", price, "\t\t", type(price))


Book title :  Basic Python 101 	 <class 'str'>
Book available :  True 		 <class 'bool'>
Book quantity :  10 		 <class 'int'>
Book price :  195.5 		 <class 'float'>


3. Perform simple arithmetic:


In [ ]:
num1 = 250
num2 = 5

sum = num1 + num2
print("num1 + num2 = ",sum)

sum = num1 - num2
print("num1 - num2 = ",sum)

sum = num1 * num2
print("num1 * num2 = ",sum)

sum = num1 / num2
print("num1 / num2 = ",sum)

num1 + num2 =  255
num1 - num2 =  245
num1 * num2 =  1250
num1 / num2 =  50.0


4. String concatenation and f-strings:


In [ ]:
First_name = "Tanatorn"
Last_name = "Palang"
Full_name = First_name + " " + Last_name
print(f"Hello, {Full_name}")

Hello, Tanatorn Palang


5. User input and type conversion:


In [ ]:
num = input("Type your biggest favourite number in the world : ")
print("Your number is", num, type(num))
result = int(num) + 5
print(f"So your lucky number is {num} + 5 = {result}")

Type your biggest favourite number in the world : 67
Your number is 67 <class 'str'>
So your lucky number is 67 + 5 = 72


### Activity 1.1: Class Discussion
- Uses of Scripting
Format: Open discussion, possibly in small groups first, then full class.
Prompt: "Based on the introduction, where do you think scripting with Python could be applied in real-world scenarios? Think about tasks you do daily or things you've seen in other courses."
Expected outcomes: Students will brainstorm and share ideas like automating file operations, simple data analysis, web development, game scripting, scientific computing, etc. Instructor will guide the discussion to highlight Python's versatility.


คำตอบ: ใช้ทำระบบอัตโนมัติลดงานที่ต้องทำซ้ำๆ เช่น จัดการ/จัดระเบียบไฟล์ ดึงข้อมูลจากเว็บต่างๆ หรือส่งอีเมลอัตโนมัติโดยไม่ต้องทำเอง
รวมถึงใช้จัดการข้อมูลจำนวนมาก เช่น Excel พล็อตกราฟรายงานวิชาเรียน และตั้งเวลาสำรองข้อมูลของระบบ เป็นต้น


### Activity 1.2: "Python Puzzlers"
- Identifying Data Types & Output
Format: Individual or pair work, followed by a class review.
Instructions: Provide a set of small Python code snippets (on a handout or projected). For each snippet, students must:
Predict the output.
Identify the data type of specific variables after certain operations.
Explain why they chose that output/data type.


In [ ]:
# Puzzler 1
x = 10
y = 3.5
z = x + y
print(z)
print(type(z))


13.5
<class 'float'>


ได้ผลลัพธ์: 13.5

ประเภท DataType : float

ทำไมถึงเป็นประเภทนี้ : เมื่อนำ data type ที่แตกต่างกันมาคำนวณกัน python จะทำการเลือก data type ที่ใหญ่ที่สุดในสมการ แล้วแปลงให้คำตอบอยู่ใน data type นั้น

In [ ]:
# Puzzler 2
name = "Alice"
age = "30" # Note: This is a string
message = name + " is " + age + " years old."
print(message)
print(type(age))

Alice is 30 years old.
<class 'str'>


ได้ผลลัพธ์: Alice is 30 years old.

ประเภท DataType : string

ทำไมถึงเป็นประเภทนี้ : python รองรับการรวม string ด้วยการใช้ "+" โดยจะนำ string ทั้งหมดมาเรียงต่อกัน

In [ ]:
# Puzzler 3
is_sunny = True
is_raining = False
print(is_sunny and is_raining)
print(type(is_sunny))

False
<class 'bool'>


ได้ผลลัพธ์: False

ประเภท DataType : boolean

ทำไมถึงเป็นประเภทนี้ : เนื่องจากในทางตรกกศาสตร์ การเชื่อมประพจน์จริงและเท็จด้วยและ จะได้ค่าความจริงเป็นเท็จ

In [ ]:
# Puzzler 4
# What happens here?
# num_str = input("Enter a number: ")
# result = num_str + 5
# print(result)

ได้ผลลัพธ์: ไม่มี

ประเภท DataType : ไม่มี

ทำไมถึงเป็นประเภทนี้ : มี comment อยู่ทุกบรรทัด